File: C++/Hard/WrongAnswer/shortest-cycle-in-a-graph_7.cpp
class Solution {
public:
    int findShortestCycle(int n, vector<vector<int>>& edges) {
        
        vector<vector<int>> graph(n);
        for (auto &edge : edges) {
            graph[edge[0]].push_back(edge[1]);
            graph[edge[1]].push_back(edge[0]);
        }
        int res = n + 1;
        for (int i = 0; i < n; ++i) {
            vector<int> dist(n, n + 1);
            dist[i] = 0;
            queue<int> q;
            q.push(i);
            while (!q.empty()) {
                int u = q.front();
                q.pop();
                for (int v : graph[u]) {
                    if (dist[v] != n + 1) {
                        if (v != i) {
                            res = min(res, dist[u] + dist[v] + 1);
                        }
                    } else {
                        dist[v] = dist[u] + 1;
                        q.push(v);
                    }
                }
            }
        }
        return res == n + 1 ? -1 : res;
    }
};

File: Java/Hard/WrongAnswer/shortest-cycle-in-a-graph_7.java
class Solution {
    public int findShortestCycle(int n, int[][] edges) {
        
        int[][] adj = new int[n][n];
        for (int[] edge : edges) {
            adj[edge[0]][edge[1]] = 1;
            adj[edge[1]][edge[0]] = 1;
        }
        int minCycle = Integer.MAX_VALUE;
        for (int[] edge : edges) {
            int[] visited = new int[n];
            int start = edge[0];
            int end = edge[1];
            Queue<int[]> queue = new LinkedList<>();
            queue.add(new int[]{start, 0});
            while (!queue.isEmpty()) {
                int[] node = queue.poll();
                if (node[0] == end) {
                    minCycle = Math.min(minCycle, node[1]);
                    break;
                }
                visited[node[0]] = 1;
                for (int i = 0; i < n; i++) {
                    if (adj[node[0]][i] == 1 && visited[i] == 0) {
                        queue.add(new int[]{i, node[1] + 1});
                    }
                }
            }
        }
        return minCycle == Integer.MAX_VALUE ? -1 : minCycle;
    }
}